pkg/operators/v1alpha1/clusterserviceversion_types.go

```diff
@@ -176,18 +177,25 @@ type WebhookDescription struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:default=443
-	ContainerPort           int32                                           `json:"containerPort,omitempty"`
-	TargetPort              *intstr.IntOrString                             `json:"targetPort,omitempty"`
-	Rules                   []admissionregistrationv1.RuleWithOperations    `json:"rules,omitempty"`
-	FailurePolicy           *admissionregistrationv1.FailurePolicyType      `json:"failurePolicy,omitempty"`
-	MatchPolicy             *admissionregistrationv1.MatchPolicyType        `json:"matchPolicy,omitempty"`
-	ObjectSelector          *metav1.LabelSelector                           `json:"objectSelector,omitempty"`
-	SideEffects             *admissionregistrationv1.SideEffectClass        `json:"sideEffects"`
-	TimeoutSeconds          *int32                                          `json:"timeoutSeconds,omitempty"`
-	AdmissionReviewVersions []string                                        `json:"admissionReviewVersions"`
+	ContainerPort  int32                                        `json:"containerPort,omitempty"`
+	TargetPort     *intstr.IntOrString                          `json:"targetPort,omitempty"`
+	Rules          []admissionregistrationv1.RuleWithOperations `json:"rules,omitempty"`
+	FailurePolicy  *admissionregistrationv1.FailurePolicyType   `json:"failurePolicy,omitempty"`
+	MatchPolicy    *admissionregistrationv1.MatchPolicyType     `json:"matchPolicy,omitempty"`
+	ObjectSelector *metav1.LabelSelector                        `json:"objectSelector,omitempty"`
+	SideEffects    *admissionregistrationv1.SideEffectClass     `json:"sideEffects,omitempty"`
```

Comment:
Note: added `omitempty` here since conversion webhooks don't need these.

Reply:
I think ports and `sideEffects` is required for webhookDescription. I remember installplan erroring when one of them were not present. Will verify and update here

Reply:
It may be worthwhile to consider having a separate conversion webhook struct in that case, and add it like `.spec.conversionwebhookdefinitions`.

Reply:
Yes, it would be helpful to have a different `conversionwebhookdefinition` struct. Because port and `conversionReviewVersions` are required for conversion WH and not for others